src/equations/ideal_glm_mhd_multiion_3d.jl

```diff
@@ -0,0 +1,1093 @@
+                                                                     electron_pressure,
+                                                                     initial_c_h)
+end
+
+# Outer constructor for `@reset` works correctly
+function IdealGlmMhdMultiIonEquations3D(gammas, charge_to_mass, electron_pressure, c_h)
+    return IdealGlmMhdMultiIonEquations3D(gammas = gammas,
+                                          charge_to_mass = charge_to_mass,
+                                          electron_pressure = electron_pressure,
+                                          initial_c_h = c_h)
+end
+
+@inline function Base.real(::IdealGlmMhdMultiIonEquations3D{NVARS, NCOMP, RealT}) where {
+                                                                                         NVARS,
+                                                                                         NCOMP,
+                                                                                         RealT
+                                                                                         }
+    RealT
+end
+
+"""
+    initial_condition_weak_blast_wave(x, t, equations::IdealGlmMhdMultiIonEquations3D)
+
+A weak blast wave (adapted to multi-ion MHD) from
+- Hennemann, S., Rueda-Ramírez, A. M., Hindenlang, F. J., & Gassner, G. J. (2021). A provably entropy 
+  stable subcell shock capturing approach for high order split form DG for the compressible Euler equations. 
+  Journal of Computational Physics, 426, 109935. [arXiv: 2008.12044](https://arxiv.org/abs/2008.12044).
+  [DOI: 10.1016/j.jcp.2020.109935](https://doi.org/10.1016/j.jcp.2020.109935)
+"""
+function initial_condition_weak_blast_wave(x, t,
+                                           equations::IdealGlmMhdMultiIonEquations3D)
+    # Adapted MHD version of the weak blast wave from Hennemann & Gassner JCP paper 2020 (Sec. 6.3)
+    # Same discontinuity in the velocities but with magnetic fields
+    RealT = eltype(x)
+    # Set up polar coordinates
+    inicenter = (0, 0, 0)
+    x_norm = x[1] - inicenter[1]
+    y_norm = x[2] - inicenter[2]
+    z_norm = x[3] - inicenter[3]
+    r = sqrt(x_norm^2 + y_norm^2 + z_norm^2)
+    phi = atan(y_norm, x_norm)
+    theta = iszero(r) ? zero(RealT) : acos(z_norm / r)
+
+    # Calculate primitive variables
+    rho = r > 0.5f0 ? one(RealT) : convert(RealT, 1.1691)
+    v1 = r > 0.5f0 ? zero(RealT) : convert(RealT, 0.1882) * cos(phi) * sin(theta)
+    v2 = r > 0.5f0 ? zero(RealT) : convert(RealT, 0.1882) * sin(phi) * sin(theta)
+    v3 = r > 0.5f0 ? zero(RealT) : convert(RealT, 0.1882) * cos(theta)
+    p = r > 0.5f0 ? one(RealT) : convert(RealT, 1.245)
+
+    prim = zero(MVector{nvariables(equations), real(equations)})
+    prim[1] = 1
+    prim[2] = 1
+    prim[3] = 1
+    for k in eachcomponent(equations)
+        set_component!(prim, k,
+                       2^(k - 1) * (1 - 2) / (1 - 2^ncomponents(equations)) * rho, v1,
```

Comment:
Does it make sense to extract this, i.e., 

`2^(k - 1) * (1 - 2) / (1 - 2^ncomponents(equations)) * rho` 

into an own variable with a descriptive name such that one has an idea what is going on here?